poethepoet/task/args.py

```diff
@@ -1,154 +1,186 @@
-        elif isinstance(args_def, dict):
-            for arg_name, params in args_def.items():
-                error = cls._validate_name(arg_name, task_name, arg_names)
-                if error:
-                    return error
-                if "name" in params:
-                    return (
-                        f"Unexpected 'name' option for arg {arg_name!r} of task "
-                        f"{task_name!r}"
-                    )
-                error = cls._validate_params(params, arg_name, task_name)
-                if error:
-                    return error
-        return None
-
-    @classmethod
-    def _validate_params(
-        cls, params: ArgParams, arg_name: str, task_name: str
-    ) -> Optional[str]:
-        for param, value in params.items():
-            if param not in arg_param_schema:
-                return (
-                    f"Invalid option {param!r} for arg {arg_name!r} of task "
-                    f"{task_name!r}"
-                )
-            if not isinstance(value, arg_param_schema[param]):
-                return (
-                    f"Invalid value for option {param!r} of arg {arg_name!r} of"
-                    f" task {task_name!r}"
-                )
-        return None
-
-    @classmethod
-    def _validate_name(
-        cls, name: Any, task_name: str, arg_names: Set[str]
-    ) -> Optional[str]:
-        if not isinstance(name, str):
-            return f"Arg name {name!r} of task {task_name!r} should be a string"
-        if not name.isidentifier():
-            return (
-                f"Arg name {name!r} of task {task_name!r} is not a valid " "identifier"
-            )
-        if name in arg_names:
-            return f"Duplicate arg name {name!r} for task {task_name!r}"
-        arg_names.add(name)
-        return None
-
-    def build_parser(self) -> argparse.ArgumentParser:
-        parser = argparse.ArgumentParser(add_help=False, allow_abbrev=False)
-        for arg in self._args:
-            parser.add_argument(
-                *arg["options"],
-                default=arg.get("default", ""),
-                dest=arg["name"],
-                required=arg.get("required", False),
-                help=arg.get("help", ""),
-            )
-        return parser
-
-    def parse(self, extra_args: Sequence[str]):
-        return vars(self.build_parser().parse_args(extra_args))
+import argparse
+from typing import (
+    Any,
+    Dict,
+    List,
+    Optional,
+    Sequence,
+    Set,
+    Tuple,
+    Type,
+    Union,
+    TYPE_CHECKING,
+)
+
+if TYPE_CHECKING:
+    from ..config import PoeConfig
+
+ArgParams = Dict[str, Any]
+ArgsDef = Union[List[str], List[ArgParams], Dict[str, ArgParams]]
+arg_param_schema: Dict[str, Union[Type, Tuple[Type, ...]]] = {
+    "default": (str, int, float, bool),
+    "help": str,
+    "name": str,
+    "options": (list, tuple),
+    "required": bool,
+    "type": str,
+}
+
+arg_type_lookup: Dict[Any, Type] = {

```

Comment:
I think there was at one point an issue I was having with mypy when I set it to str oddly enough. But In practice absolutely this would be a better solution. I'll switch it back and take a look at what was going on in mypy in more detail